presto-parquet/src/main/java/com/facebook/presto/parquet/reader/ParquetReader.java

```diff
@@ -627,7 +635,24 @@ private static Block rewriteLongArrayBlock(LongArrayBlock longArrayBlock, Type t
                 newBlockBuilder.appendNull();
             }
             else {
-                targetType.writeLong(newBlockBuilder, longArrayBlock.getLong(position, 0));
+                targetType.writeLong(newBlockBuilder, longArrayBlock.getLong(position));
+            }
+        }
+
+        return newBlockBuilder.build();
+    }
+
+    private static Block rewriteTimeLongArrayBlock(LongArrayBlock longArrayBlock, Type targetType)
+    {
+        int positionCount = longArrayBlock.getPositionCount();
+        BlockBuilder newBlockBuilder = targetType.createBlockBuilder(null, positionCount);
+        for (int position = 0; position < positionCount; position++) {
+            if (longArrayBlock.isNull(position)) {
+                newBlockBuilder.appendNull();
+            }
+            else {
+                // We shift the bits to encode timezone information
+                targetType.writeLong(newBlockBuilder, DateTimeEncoding.packDateTimeWithZone(longArrayBlock.getLong(position), UTC_KEY));
```

Comment:
We shouldn't need to encode TZ information, but rather just mark the type as being adjusted to UTC?